cmd/web.go

```diff
@@ -91,6 +111,9 @@ func runWeb(ctx *cli.Context) error {
 	case setting.HTTP:
 		err = runHTTP(listenAddr, context2.ClearHandler(m))
 	case setting.HTTPS:
+		if setting.RedirectPort80 {
+			go runHTTPRedirector()
```

Comment:
Don't fire-n-forget

Reply:
The others don't listen in goroutines, so when they error out the whole process errors out. The use of a goroutine here means if/when that listener errors out, it does so silently and without any mechanism for restarting.

Reply:
Looking around it seems like we don't actually track our long-running goroutines 🙄

Reply:
Having a loose goroutine like this is risky at best. And will end up with "this **** is broken"-issues that's only solvable by creating a service-handler for tracking long-running goroutines. Which is goind to be a pain :(

Reply:
Yes I understand that goroutines are threads.  And a thread hanging or crashing will cause the service to go away.  However if we were to write a thread manager it would presumably required re-engineering the current "main thread" that holds the main gitea web server as well.  Even if we did do this you have a similar problem with the actual thread manager.  What if it hangs for example and your threads crash for independent reasons?

I don't agree with your assessment that the solution I have chosen _for the given feature_ is risky.  It is a exactly 5 lines of code being executed by the redirector service.  How much crashing or hanging are we expecting to come out of this?  And even if the redirector thread does happen to hang should it be restarted?  I suppose something like a DDOS could bring it down, but that seems a stretch.  If someone was trying to bring down your site it seems unlikely that they would only be interested in crashing your port 80 to 443 redirector and not attacking your main site on port 443.  At a certain point, with these kinds of problems, you need outside tools like pingdom or uptimerobot.

So where does this leave this PR?  I've mostly given up on this and accepted that I have to run gitea behind Apache acting as a proxy.  It seems odd to have to use a heavyweight server in front of the nice, lightweight gitea web server.  If you can give me some specific suggestions about what adjustments are needed to satisfy your concerns please do.  Or if you think the feature is not needed, or impossible to implement without radical re-engineering of the main gitea web server, then please close the PR.